### PR TITLE
Protect exploits from cheaters.

### PR DESCRIPTION
**Pull request by Misha0717**
Removed all the client sided 'discord logs' logic and also created server sided checks on cheaters instead client sided.

### CODE REVIEW

File: server.lua [hunk 1]
```diff
@@ -89,9 +126,32 @@ RegisterServerEvent('dream-postal:server:log', function(data)
...
+        currentWorkers[source] = nil
+    end
+
+    TriggerEvent("dream-postal:server:log", { source = source, type = "success", message = t('ended_shift')})
+end)
+
 
 -- utils
 
+function IsNearAnyDeliverPoint(source)
```
Comment (kalekaneki) [inline server.lua]: Not a showstopper, I like lowercase convention to differentiate from natives doesn't need to be done in this pr i can do us a follow up commit

```suggestion
function isNearAnyDeliverPoint(source)
```

**Comment by kalekaneki** [comment]
I think we need to remove this snippet of code since we are no longer using the `log` function client side

```lua
---@param data { type: 'success' | 'error', message: string }
function log(data)
    local type = data.type
    local message = data.message

    local logData = {
        type = type,
        message = message,
        postalJobState = (type == 'error') and postalJobState or nil,
    }

    TriggerServerEvent('dream-postal:server:log', logData)
end
```

File: server.lua [hunk 2]
```diff
@@ -46,33 +99,17 @@ local DISCORD_NAME = "DreamLife RP BOT"
 local DISCORD_IMAGE = "https://img.freepik.com/premium-vector/cute-robot-waving-hand-cartoon-illustration_138676-2744.jpg?w=2000"
 local LOG_FOOTER = '[DreamLife RP LOGS]'
 
----@param data { type: string, message: string, postalJobState?: table }
-RegisterServerEvent('dream-postal:server:log', function(data)
-    local src = source
+---@param data { type: string, message: string, postalJobState?: table, source: int }
```
Comment (kalekaneki) [inline server.lua]: I don't believe we're passing postalJobState to this function anymore

```suggestion
---@param data { type: string, message: string, source: int }
```

File: server.lua [hunk 3]
```diff
@@ -89,9 +126,32 @@ RegisterServerEvent('dream-postal:server:log', function(data)
...
+    TriggerEvent("dream-postal:server:log", { source = source, type = "success", message = t('ended_shift')})
+end)
+
 
 -- utils
 
+function IsNearAnyDeliverPoint(source)
+    local coords = GetEntityCoords(GetPlayerPed(source))
+
+    for k,v in pairs(Config.POSTAL_DROP_OFF_PACKAGE) do
```
Comment (kalekaneki) [inline server.lua]: not a showstopper, up to u if u want this to be done in scope of this pr

```suggestion
    for _, v in pairs(Config.POSTAL_DROP_OFF_PACKAGE) do
```

**Review by kalekaneki — COMMENTED** [review]
only one showstopper which is removing the log function. Rest lgtm 🚀 

This is dope my first time reviewing a pr from someone not from work 🤓

**Comment by Misha0717** [comment]
yeah you are right on all the changes 😂 I created a commit to fix all these things. 👌

**Comment by kalekaneki** [comment]
Code looks amazing

Now I just need to run this on my server double check all functionally works and I'll merge your pr ✨

**Comment by Misha0717** [comment]
<3

**Review by kalekaneki — APPROVED** [review]
🚀

**Pull request merged**